operator/src/main/java/org/keycloak/operator/v2alpha1/KeycloakDeployment.java

```diff
@@ -146,9 +151,206 @@ private void addInitContainer(Deployment baseDeployment, List<String> extensions
         baseDeployment.getSpec().getTemplate().getSpec().setInitContainers(Collections.singletonList(initContainer));
     }
 
+    public void validatePodTemplate(KeycloakStatusBuilder status) {
```

Comment:
- `image` -> is a top level feature and is not overridden by the `podTemplate` but added to the validation
 - `replicas` -> is a property of the "Deployment"

Am I missing any other field?
Do you want to emit a warning even for fields that are appropriately merged? This would become hard to maintain for no big UX advantage I believe.

Reply:
Ah, right, `replicas` are not set in PodTemplate.

The idea is to emit a warning if user tries to modify a field that has a dedicated counterpart in CR via the PodTemplates. Currently only `image`.